### PR TITLE
Refactor validate method to accept a CDAP Schema JSON object

### PR DESCRIPTION
**Pull request by abhinavbansal246**
Refactor validate Api of data dictionary to

* Accept cdap schema json 
* validate more than 1 column in a single call

### CODE REVIEW

File: src/main/java/co/cask/tracker/DataDictionaryHandler.java [hunk 1]
```diff
@@ -238,21 +240,39 @@ public void validate(HttpServiceRequest request, HttpServiceResponder responder)
...
       return;
     }
-    DictionaryResult fromTable = createDictionaryResultFromRow(row);
-    result = fromTable.validate(fromSchema);
-    if (result.isEmpty()) {
+
+    List<DictionaryResult> dictionaryResults = getDictionaryResultsfromSchema(recordSchema);
+    List<Map> results = new ArrayList<>();
+    for (DictionaryResult fromSchema: dictionaryResults) {
+      Row row = dataDictionaryTable.get(new Get(fromSchema.getColumnName().toLowerCase()));
```
Comment (russorat) [inline src/main/java/co/cask/tracker/DataDictionaryHandler.java]: does this handle names like user_id vs userId vs UserID?

File: src/main/java/co/cask/tracker/DataDictionaryHandler.java [hunk 2]
```diff
@@ -238,21 +240,39 @@ public void validate(HttpServiceRequest request, HttpServiceResponder responder)
...
+      Map<String, Object> result = new LinkedHashMap<>();
+      if (row.isEmpty()) {
+        result.put("columnName", fromSchema.getColumnName());
```
Comment (russorat) [inline src/main/java/co/cask/tracker/DataDictionaryHandler.java]: extract to global static variable?

File: src/main/java/co/cask/tracker/DataDictionaryHandler.java [hunk 3]
```diff
@@ -238,21 +240,39 @@ public void validate(HttpServiceRequest request, HttpServiceResponder responder)
...
+        List<String> reasonList = new ArrayList<>();
+        reasonList.add("The column does not exist in the data dictionary.");
```
Comment (russorat) [inline src/main/java/co/cask/tracker/DataDictionaryHandler.java]: extract to global static variable?

File: src/main/java/co/cask/tracker/DataDictionaryHandler.java [hunk 4]
```diff
@@ -238,21 +240,39 @@ public void validate(HttpServiceRequest request, HttpServiceResponder responder)
...
+        result.put("reason", reasonList);
```
Comment (russorat) [inline src/main/java/co/cask/tracker/DataDictionaryHandler.java]: extract to global static variable?

**Comment by russorat** [comment]
minor comments. otherwise LGTM

File: src/main/java/co/cask/tracker/DataDictionaryHandler.java (hunk 1 above)
Comment (abhinavbansal246) [inline src/main/java/co/cask/tracker/DataDictionaryHandler.java]: yes, Adding a unit test to verify that.

File: src/main/java/co/cask/tracker/DataDictionaryHandler.java (hunk 2 above)
Comment (abhinavbansal246) [inline src/main/java/co/cask/tracker/DataDictionaryHandler.java]: Each column to validate will have its own result object (also true for reason list) that means we need to initialize a new hashmap everytime we encounter a new dictionary result object. Even if we create a static object, we will have to initialize a new hashmap for every column in the query

File: src/main/java/co/cask/tracker/DataDictionaryHandler.java (hunk 3 above)
Comment (abhinavbansal246) [inline src/main/java/co/cask/tracker/DataDictionaryHandler.java]: same as result object

**Pull request merged**